homeassistant/components/climate/generic_thermostat.py

```diff
@@ -124,26 +139,31 @@ def __init__(self, hass, name, heater_entity_id, sensor_entity_id,
             async_track_time_interval(
                 hass, self._async_keep_alive, self._keep_alive)
 
-        sensor_state = hass.states.get(sensor_entity_id)
-        if sensor_state:
-            self._async_update_temp(sensor_state)
-
     @asyncio.coroutine
     def async_added_to_hass(self):
         """Run when entity about to be added."""
         # Check If we have an old state
         old_state = yield from async_get_last_state(self.hass,
                                                     self.entity_id)
         if old_state is not None:
-            # If we have no initial temperature, restore
-            if self._target_temp is None:
-                self._target_temp = float(
-                    old_state.attributes[ATTR_TEMPERATURE])
+            self._target_temp = float(
```

Comment:
The target temperature should only be restored from a previous state if no initial value (i.e., `CONF_TARGET_TEMP`) was given. The previous logic
```
if self._target_temp is None:
    self._target_temp = …
```
was correct.

Reply:
I have to disagree from a usage perspective. This use case covers an accidental reboot of your unit, by keeping your previously set temperature. If you for example manually set a low temperature because you're away from home and your unit reboots, you will get probably a default for when you're at home. The more complex your schedule to update the temperature the more different will be your set default temperature from a restored one. 

The other way around this would be to have something to set a target_temp after you try restoring it from state, and avoid having something set in settings.

Reply:
Setting an initial value (for the target temperature / operation mode / whatever) has only one use case: You want this value to be set every time your HA instance starts, be it after a power loss or a manual restart. If that's not what you want, just don't set an initial value (which will then always restore the previous state). Thus, restoring the previous state even if an initial value is present basically renders the initial value useless.

Reply:
I guess that makes sense. I was looking at config value as "initial" value when you have nothing saved to avoid having your target temperature as None. 

The UI control doesn't look that well with "None" temperature and when starting to press arrows you start from 0 degrees regardless of your min/max settings. 

Assuming everyone looks at the config value as an "always fallback" value after restart if set, I guess it would require some fixes on the UI to make the setup with "None" work better.

Reply:
Yes, that's true. If you don't want an initial value to always be set on every restart, you need to configure your entity once (after you first added it) "starting from scratch". The UI should support this.

Reply:
Didn't think of that option.